src/preferences/upgrade.cpp

```diff
@@ -515,36 +577,46 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
     }
 
     const auto configFileVersion = QVersionNumber::fromString(configVersion);
+    // Only update the framerate if the version we are migrating from is less than 2.4.0.
+    if (configFileVersion < QVersionNumber(2, 4, 0)) {
+        config->set(ConfigKey("[Waveform]", "FrameRate"), ConfigValue(60));
+    }
 
-    // When upgrading from 2.3.x or older to 2.4, or when upgrading
-    // from 2.4.0-beta once we are out of beta
-    if (QVersionNumber::fromString(configVersion) < QVersionNumber(2, 4, 0) ||
-            (VersionStore::version() != "2.4.0-beta" &&
-                    configVersion.startsWith("2.4.0-"))) {
+    // When upgrading from 2.3.x or older to 2.5, or when upgrading
+    // from 2.5.0-beta once we are out of beta
+    if (configFileVersion < QVersionNumber(2, 5, 0) ||
+            (VersionStore::version() != "2.5.0-beta" &&
+                    configVersion.startsWith("2.5.0-"))) {
         // Proactively move users to an all-shader waveform widget type and set the
         // framerate to 60 fps
-        bool ok = false;
-        auto waveformType =
-                config->getValueString(ConfigKey("[Waveform]", "WaveformType"))
-                        .toInt(&ok);
-        if (ok) {
-            config->set(ConfigKey("[Waveform]", "WaveformType"),
-                    ConfigValue(upgradeToAllShaders(
-                            static_cast<WaveformWidgetType::Type>(
-                                    waveformType))));
-        }
-        config->set(ConfigKey("[Waveform]", "FrameRate"), ConfigValue(60));
-
+        int waveformType =
+                config->getValue<int>(ConfigKey("[Waveform]", "WaveformType"));
+        // values might be out of range for the enum, avoid undefined
+        // behavior by not casting to the enum type just yet.
+        int waveformBackend = config->getValue<int>(
+                ConfigKey("[Waveform]", "use_hardware_acceleration"));
+        int waveformOption = config->getValue<int>(ConfigKey("[Waveform]", "waveform_options"));
+        auto [correctedWaveformType,
```

Comment:
I think this needs more work now that this has slipped to 2.6. 

I think we need to keep the old 2.4 code and then add this for 2.6. Whats your opinion @daschuer?

Reply:
yes I do think we bumped this to 2.6